utilities/test_suite/HIP_NEW/Tensor_hip.cpp

```diff
@@ -0,0 +1,2406 @@
+            else
+                missingFuncFlag = 1;
+
+            break;
+        }
+        default:
+            missingFuncFlag = 1;
+            break;
+        }
+
+        hipDeviceSynchronize();
+        end = clock();
+        end_omp = omp_get_wtime();
+
+
+        if (missingFuncFlag == 1)
+        {
+            printf("\nThe functionality %s doesn't yet exist in RPP\n", func.c_str());
+            return -1;
+        }
+
+        gpu_time_used = ((double)(end - start)) / CLOCKS_PER_SEC;
+        omp_time_used = end_omp - start_omp;
+
+        if (gpu_time_used > max_time_used)
+            max_time_used = gpu_time_used;
+        if (gpu_time_used < min_time_used)
+            min_time_used = gpu_time_used;
+        avg_time_used += gpu_time_used;
+    }
+
+    if (test_type == 0)
+    {
+        gpu_time_used = gpu_time_used;
+        omp_time_used = omp_time_used;
+        cout << "\nGPU Time - Tensor : " << gpu_time_used;
+        cout << "\nOMP Time - Tensor : " << omp_time_used;
+        printf("\n");
+    }
+    else
+    {
+        avg_time_used /= num_iterations;
+        max_time_used = max_time_used;
+        min_time_used = min_time_used;
+        avg_time_used = avg_time_used;
+        // Display measured times
+
+        cout << fixed << "\nmax,min,avg in ms = " << max_time_used << "," << min_time_used << "," << avg_time_used << endl;
+    }
+
+    if (test_type == 0)
+    {
+        // Reconvert other bit depths to 8u for output display purposes
+        string fileName = std::to_string(ip_bitDepth);
+        ofstream outputFile(fileName + ".csv");
+        ofstream outFile(func + ".csv");
+
+        if (ip_bitDepth == 0)
+        {
+            hipMemcpy(output, d_output, oBufferSizeInBytes_u8, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp;
+            outputTemp = output + dstDescPtr->offsetInBytes;
+
+            if (outputFile.is_open())
+            {
+                for (int i = 0; i < oBufferSize; i++)
+                {
+                    outputFile << (Rpp32u) *outputTemp << ",";
+                    outFile << (Rpp32u) *outputTemp << ",";
+                    outputTemp++;
+                }
+                outputFile.close();
+            }
+            else
+                cout << "Unable to open file!";
+        }
+        else if ((ip_bitDepth == 1) || (ip_bitDepth == 3))
+        {
+            hipMemcpy(outputf16, d_outputf16, oBufferSizeInBytes_f16, hipMemcpyDeviceToHost);
+            Rpp8u *outputTemp;
+            outputTemp = output + dstDescPtr->offsetInBytes;
+            half *outputf16Temp;
+            outputf16Temp = (half *)((Rpp8u *)outputf16 + dstDescPtr->offsetInBytes);
+
+            if (outputFile.is_open())
+            {
+                for (int i = 0; i < oBufferSize; i++)
+                {
+                    outputFile << (char) *outputf16Temp << ",";
+                    *outputTemp = (Rpp8u)RPPPIXELCHECK((float)*outputf16Temp * invConversionFactor);
```

Comment:
If these csv files written here are being used for the final pixelwise compare, we cannot use RPPPIXELCHECK and Rpp8u for the float versions

Reply:
Done